Bottom: 43fe26892289ed67b67c6d60e767a25ed5d67b41
Top:    ea178366eee14b5550d35c155af48676d8a64d8a
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-05 14:16:16 -0500

Create new role buildbot-worker-assets

Starts a web server for distributing assets to buildbot workers.


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index e14b93b9..cb94b281 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -21,6 +21,19 @@
       new_user_email: buildbot@novatech-llc.com
       new_user_pass: "{{ lookup('password', '/dev/null') }}"
 
+- name: Setup Buildbot worker system
+  hosts: builder
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  - buildbot-worker
+  vars:
+    buildbot_master: "{{ hostvars.buildbot_master.ip_addr }}"
+  roles:
+  - role: buildbot-worker-assets
+
 - name: Gitlab public keys for buildbot containers
   hosts: daedalus
   become: true
diff --git a/roles/buildbot-worker-assets/tasks/main.yml b/roles/buildbot-worker-assets/tasks/main.yml
new file mode 100644
index 00000000..df9bda29
--- /dev/null
+++ b/roles/buildbot-worker-assets/tasks/main.yml
@@ -0,0 +1,16 @@
+---
+- name: Start asset server
+  docker_container:
+    name: builder-assets
+    image: "httpd:alpine"
+    volumes:
+    - "/mnt/assets:/usr/local/apache2/htdocs/:ro"
+    ports:
+    - 80
+  register: asset_server
+
+- set_fact:
+    buildbot_asset_host: "http://\
+    {{asset_server.ansible_facts.docker_container.NetworkSettings.IPAddress}}\
+    :{{asset_server.ansible_facts.docker_container.NetworkSettings.Ports['80/tcp'][0].HostPort}}\
+    "
